CHANGELOG.md

```diff
@@ -1,3 +1,23 @@
+## [4.0.0] - 2023-07-28
```

Comment:
FYI: We haven't made any releases with Hive3 mostly because internally we're not ready to move to hive3 just yet. 
We'll have to change either the artifact id or version to reflect hive3 otherwise we'll get clashes if make a release of the main branch. 
Are you ok to build your own version based on the branch for now?

Reply:
No need to merge with master. But can merge with hive-3.x version

Reply:
FYI. All the integration tests are working fine